Metaverse/Source/Metaverse/Private/NPC/CallDynamicContract.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022, Cronos Labs. All Rights Reserved
+
+#include "NPC/CallDynamicContract.h"
+#include "Components/PrimitiveComponent.h"
+#include "Components/WalletConnectTriggerComponent.h"
+#include "../../MetaverseCharacter.h"
+
+// Called when the game starts or when spawned
+void ACallDynamicContract::BeginPlay() {
+    Super::BeginPlay();
+    GetWalletConnectTriggerComponent()->OnComponentBeginOverlap.AddDynamic(
+        this, &ACallDynamicContract::OnCallDynamicContractBeginOverlap);
+
+    // connect member function to callback variable
+    CallDynamicContractCallback.BindDynamic(
+        this, &ACallDynamicContract::OnCallDynamicContract);
+}
+
+void ACallDynamicContract::OnCallDynamicContractBeginOverlap(
+    UPrimitiveComponent *OverlappedComponent, AActor *OtherActor,
+    UPrimitiveComponent *OtherComp, int32 OtherBodyIndex, bool bFromSweep,
+    const FHitResult &SweepResult) {
+
+    AMetaverseCharacter *MetaverseCharacter =
+        Cast<AMetaverseCharacter>(OtherActor);
+    if (MetaverseCharacter != nullptr) {
+        if (UserDynamicContractObject == nullptr) {
+            bool success = false;
+            FString output_message = "";
+            UserDynamicContractObject =
+                MetaverseCharacter->CreateDynamicContract(
+                    UserContractAddress, UserContractAbiJson, success,
```

Comment:
Since UserContractAbiJson is not supposed to be empty, If set in the editor, where is the AbiJson file? How about include the Abi file into the demo? I guess if the abi file is not included in the demo, the demo would not run on other computers.

The same applies on UserContractAddress, UserFunctionName, UserFunctionArgs. 
Please provide the default value of those variables on C++ instead of setting the default values on editor.